src/kademlia/dht_storage.cpp

```diff
@@ -83,17 +84,15 @@ namespace
 	struct dht_immutable_item
 	{
 		// the actual value
-		std::unique_ptr<char[]> value;
+		std::vector<char> value;
```

Comment:
I'm surprised that `libtorrent::buffer` can overallocate, not sure if it's a good fit for this case